net/openssh/files/sshd.init

```diff
@@ -6,8 +6,351 @@ STOP=50
+	# approximately 45 seconds.  Setting a zero
+	# ClientAliveCountMax disables connection termination.
+	#
+	# Therefore, to mimic IdleTimeout from dropbear, we set
+	# ClientAliveCountMax to 1 and ClientAliveInterval to
+	# the time we want to wait
+	if [ "${IdleTimeout}" -ne 0 ]; then
+		append_config "ClientAliveCountMax" "1"
+		append_config "ClientAliveInterval" "${IdleTimeout}"
+	fi
+}
+
+publish_ssh_service_over_mdns()
+{
+	local mdns="${1}"
+	local Port="${2}"
+
+	if [ "${mdns}" -ne 0 -a "${Port}" -gt 0 ]; then
+		procd_add_mdns "ssh" "tcp" "${Port}" "daemon=sshd"
+	fi
+}
+
+set_max_auth_tries()
+{
+	local MaxAuthTries="${1}"
+
+	if [ "${MaxAuthTries}" -gt 0 ]; then
+		append_config "MaxAuthTries" "${MaxAuthTries}"
+	fi
+}
+
+set_root_login()
+{
+	local RootLogin="${1}"
+	local RootPasswordAuth="${2}"
+
+	if [ "${RootLogin}" -eq 0 ]; then
+		append_config "PermitRootLogin" "no"
+	else
+		if [ "${RootPasswordAuth}" -eq 0 ]; then
+			append_config "PermitRootLogin" "prohibit-password"
+		else
+			append_config "PermitRootLogin" "yes"
+		fi
+	fi
+}
+
+set_password_auth()
+{
+	local PasswordAuth="${1}"
+
+	if [ "${PasswordAuth}" -eq 0 ]; then
+		append_config "PasswordAuthentication" "no"
+	else
+		append_config "PasswordAuthentication" "yes"
+	fi
+}
+
+set_params()
+{
+	local ConfigFile="${1}"
+
+	# start fresh
+	rm -rf "${TEMPCONF}"
+	touch "${TEMPCONF}"
+
+	set_allow_users "${AllowUsers}"
+	set_banner_file "${BannerFile}"
+	set_ciphers "${Ciphers}"
+	set_gateway_ports "${GatewayPorts}"
+	set_host_key_algorithms "${HostKeyAlgorithms}"
+	set_host_key_files "${HostKeyFiles}"
+	set_idle_timeout "${IdleTimeout}"
+	set_kex_algorithms "${KexAlgorithms}"
+	set_listen_addresses "${Port}" "${IpAddrs}"
+	set_mac_algorithms "${MacAlgorithms}"
+	set_max_auth_tries "${MaxAuthTries}"
+	set_password_auth "${PasswordAuth}"
+	set_root_login "${RootLogin}" "${RootPasswordAuth}"
+
+	# finalize
+	cp "${TEMPCONF}" "${ConfigFile}"
+}
+
+sshd_instance()
+{
+	local IpAddrs
+	local Cfg="$1"
+	local ValidationResult="${2}"
+
+	[ "${ValidationResult}" = 0 ] || {
+		echo "validation failed"
+		return 1
+	}
+
+	[ "${enable}" -eq 0 ] && return 0
+
+	[ -n "${Interface}" ] && {
+		network_get_ipaddrs_all IpAddrs "${Interface}" || {
+			echo "interface ${Interface} has no physdev or physdev has no suitable ip"
+			return 1
+		}
+	}
+
+	local PidFile="/var/run/${NAME}.${Cfg}.pid"
+	local ConfigFile="/etc/ssh/${NAME}.${Cfg}.conf"
+
+	# create config file
+	set_params "$ConfigFile"
+
+	# set up procd instance
+	procd_open_instance $Cfg
+	procd_set_param command $PROG -D
+	procd_append_param command -o "PidFile $PidFile"
+
+	# pass config to daemon
+	procd_append_param command -f "$ConfigFile"
+
+	# announce mdns service
+	publish_ssh_service_over_mdns "${mdns}" "${Port}"
+
+	procd_set_param respawn
+	procd_close_instance
+}
+
+# for adding trigger
+load_interfaces()
+{
+	config_get Interface "$1" Interface
+	config_get enable "$1" enable 1
```

Comment:
config_get_bool enable "$1" enable 1

Reply:
Hello, kindly review:

- added a check for both enable and enabled
- updated to use config_get_bool instead of config_get